src/compas/scene/scene.py

```diff
@@ -6,58 +6,95 @@
 from .context import clear
 
 
+class SceneObjectNode(TreeNode):
+    def __init__(self, sceneobject):
+        super(SceneObjectNode, self).__init__(name=sceneobject.name)
+        self.object = sceneobject
+
+    @property
+    def parent_object(self):
+        if self.parent and isinstance(self.parent, SceneObjectNode):
+            return self.parent.object
+        return None
+
+    @property
+    def children_objects(self):
+        return [child.object for child in self.children]
+
+    def add_item(self, item, **kwargs):
+        sceneobject = SceneObject(item, **kwargs)
+        node = SceneObjectNode(sceneobject)
+        self.add(node)
+        sceneobject._node = node
+        return sceneobject
+
+
+class SceneTree(Tree):
+    def __init__(self, name=None):
+        super(SceneTree, self).__init__(name=name)
+        root = TreeNode(name="root")
+        self.add(root)
+
+    @property
+    def objects(self):
+        return [node.object for node in self.nodes if isinstance(node, SceneObjectNode)]
+
+    def add_object(self, sceneobject, parent=None):
+        node = SceneObjectNode(sceneobject)
+        if parent is None:
+            self.add(node, parent=self.root)
+        else:
+            parent_node = self.get_node_from_object(parent)
+            self.add(node, parent=parent_node)
+
+        sceneobject._node = node
+
+    def get_node_from_object(self, sceneobject):
+        for node in self.nodes:
+            if isinstance(node, SceneObjectNode):
+                if node.object == sceneobject:
+                    return node
+        raise ValueError("Scene object not in scene")
+
+
```

Comment:
Adding these two classes to abstract tree operations away a bit from the `Scene`